### PR TITLE
Add accessibilityAutoFocus to Text component

### PR DESCRIPTION
**Pull request by henrytao-me**
Follow up of https://github.com/cds-snc/covid-shield-mobile/pull/534

### CODE REVIEW

File: src/components/Text.tsx [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
-const BaseText = (props: TextProps, ref?: React.LegacyRef<any>) => {
+export const Text = ({accessibilityAutoFocus = false, ...props}: TextProps) => {
   const styledProps = useRestyle(textRestyleFunctions, props);
-  return <RNText {...styledProps} ref={ref} />;
+  const autoFocusRef = useAccessibilityAutoFocus(accessibilityAutoFocus);
+  return <RNText accessible ref={autoFocusRef} {...styledProps} />;
 };
 
-export const Text = forwardRef(BaseText);
```
Comment (henrytao-me) [inline src/components/Text.tsx:15]: There is no need for forwardRef when we use prop.

**Comment by nickmly** [comment]
Works great on Android!

File: src/shared/useAccessibilityAutoFocus.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+const AUTO_FOCUS_DELAY = 200;
+
+/**
+ * Remember to add `accessible={true}` to target component
+ */
+export const useAccessibilityAutoFocus = (isActive = true) => {
+  const [autoFocusRef, setAutoFocusRef] = useState<any>();
+  useLayoutEffect(() => {
+    if (!autoFocusRef || !isActive) {
```
Comment (henrytao-me) [inline src/shared/useAccessibilityAutoFocus.ts:18]: First, we don't want to do this on unnecessary components.

File: src/shared/useAccessibilityAutoFocus.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      return;
+    }
+    const {callable, cancelable} = createCancellableCallbackPromise(
+      () => AccessibilityInfo.isScreenReaderEnabled().then(delay(AUTO_FOCUS_DELAY)),
```
Comment (henrytao-me) [inline src/shared/useAccessibilityAutoFocus.ts:22]: Second, we only do the timeout and focus if screenReader is enabled.

Comment (henrytao-me) [reply src/shared/useAccessibilityAutoFocus.ts:22]: Adding the timeout 200ms seems to be more stable. `createCancellableCallbackPromise` will make sure if component is unmounted before timeout is finished, it won't leak.

**Pull request merged**